src/pcre2_study.c

```diff
@@ -921,6 +921,138 @@ if (table_limit != 32) for (c = 24; c < 32; c++) re->start_bitmap[c] = 0xff;
 
 
 
+#if defined SUPPORT_UNICODE && PCRE2_CODE_UNIT_WIDTH == 8
+/*************************************************
+*     Set starting bits for a character list.    *
+*************************************************/
+
+/* This function sets starting bits for a character list. It enumerates
+all characters and character ranges in the character list, and sets
+the starting bits accordingly.
+
+Arguments:
+  code           pointer to the code
+  start_bitmap   pointer to the starting bitmap
+
+Returns:         nothing
+*/
+static void
+study_char_list(PCRE2_SPTR code, uint8_t *start_bitmap)
+{
+uint32_t type, list_ind;
+uint32_t char_list_add = XCL_CHAR_LIST_LOW_16_ADD;
+uint32_t range_start = ~(uint32_t)0, range_end = 0;
+uint8_t *next_char;
+PCRE2_UCHAR start_buffer[6], end_buffer[6];
+PCRE2_UCHAR start, end;
+
+/* Only needed in 8-bit mode at the moment. */
+type = (uint32_t)(code[0] << 8) | code[1];
+code += 2;
+
+/* Align characters. */
+next_char = (uint8_t*)code;
```

Comment:
The PCRE2_SPTR is uses mode dependent characters, I need to know the exact type here, and byte is useful for aligning as well (it is not nice to store an unaligned pointer).

Reply:
FWIW, in THIS function PCRE2_SPTR is always a pointer to bytes, which you could even make "const", so here it would work eitherway, but I see why it is better to be explicit after I went wild thinking I might have found a suggestion to improve the code ;)

Reply:
I forgot this is the study code. I don't want to change it for consistency with other codes. Adding const could work though.